releasenotes/notes/deprecate-self-connection-665647384ae2792b.yaml

```diff
@@ -0,0 +1,4 @@
+---
+deprecations:
+  - |
+    Deprecate connecting a Component to itself when calling `Pipeline.connect()`, it will raise in version `2.7.0`
```

Comment:
```suggestion
    Deprecate connecting a Component to itself when calling `Pipeline.connect()`, it will raise an error from version `2.7.0` onwards
```